homeassistant/components/shelly/entity.py

```diff
@@ -131,12 +138,27 @@ async def async_restore_block_attribute_entities(
             continue
 
         attribute = entry.unique_id.split("-")[-1]
-        description = BlockAttributeDescription(
-            name="",
-            icon=entry.original_icon,
-            unit=entry.unit_of_measurement,
-            device_class=entry.original_device_class,
+        description: BlockAttributeDescription | NumberAttributeDescription | None = (
+            None
         )
+        if entry.capabilities:
+            description = NumberAttributeDescription(
+                name="",
+                icon=entry.original_icon,
+                unit=entry.unit_of_measurement,
+                device_class=entry.original_device_class,
+                min=cast(float, entry.capabilities.get("min")),
+                max=cast(float, entry.capabilities.get("max")),
+                step=cast(float, entry.capabilities.get("step")),
+                mode=cast(NumberMode, entry.capabilities.get("mode")),
+            )
+        else:
+            description = BlockAttributeDescription(
+                name="",
+                icon=entry.original_icon,
+                unit=entry.unit_of_measurement,
+                device_class=entry.original_device_class,
+            )
```

Comment:
Maybe this should be passed as a `Callable[[RegistryEntry], BlockAttributeDescription]` so that the NumberAttributeDescription doesn't have to be inside `entity.py`